### PR TITLE
Accept states create player endpoint

### PR DESCRIPTION
**Pull request by Sarahtonein**

### CODE REVIEW

File: publicModels/state.go [hunk 1]
```diff
@@ -6,7 +6,7 @@ type State struct {
 	// publicModels.State holds public fields for the State model.
 
 	gorm.Model         // NOTE: COMEBACK: Accepting ID from Client Should only be for debug mode
-	StateID    uint    `json:"stateID"`
+	StateID    uint    `json:"ID"`
```
Comment (Balugrizzly) [inline publicModels/state.go:9]: Why is this changed?

Comment (Sarahtonein) [reply publicModels/state.go:9]: > Why is this changed?

Changed as the JSON sent to the client displayed as stateID but documentation has ID:

```
  "states": [
    {
      "id": 1,
      "value": 0.4
    },
    {
      "id": 2,
      "value": 0.1
    }
  ]
}
```

Comment (Balugrizzly) [reply publicModels/state.go:9]: This will conflict with gorm.Model.ID probably.
And by extension with a very high likely hood break all the other areas that interact with state 
The documentation is accurate and correct and does not justify your change.
The "ID" is comming from the embedded gorm.Model.ID the stateID is something different that is omitted in the readme example.
Which is wanted behavior in case that would be a empty field.
If you implement something that changes the functionality that by definition contradicts the documentation.
Documentation is not always the source of truth when implementing. And its very common to document your implementation after you implemented it. <- I know this is a bit confusing and you might be getting mixed signals on what to trust feel free to bring it up in voice

Comment (Sarahtonein) [reply publicModels/state.go]: > This will conflict with gorm.Model.ID probably. And by extension with a very high likely hood break all the other areas that interact with state The documentation is accurate and correct and does not justify your change. The "ID" is comming from the embedded gorm.Model.ID the stateID is something different that is omitted in the readme example. Which is wanted behavior in case that would be a empty field. If you implement something that changes the functionality that by definition contradicts the documentation. Documentation is not always the source of truth when implementing. And its very common to document your implementation after you implemented it. <- I know this is a bit confusing and you might be getting mixed signals on what to trust feel free to bring it up in voice

Yeah I think you are correct on the conflict, I have changed this back.

Example request body with all accepted fields:


```
{
  "UserID": "33b7e1f3-6f8e-40b9-97dc-c54d9162vb05",
  "Name": "User1",
  "Layer": "layer1",
  "Ens": "",
  "Active": true,
  "Transform": {
    "Position": {
      "x": 1,
      "y": 2,
      "z": 3
    },
    "Rotation": {
      "x": 4,
      "y": 5,
      "z": 6
    },
    "Scale": {
      "x": 7,
      "y": 8,
      "z": 9
    }
  },
  "states": [
    {
      "id": 1,
      "value": 0.4
    },
    {
      "id": 2,
      "value": 0.1
    }
  ]
}
```

Response:

```
{
	"ID": 22,
	"CreatedAt": "2023-10-23T01:53:26.990111438Z",
	"UpdatedAt": "2023-10-23T01:53:26.990111438Z",
	"DeletedAt": null,
	"UserID": "c9610fc0-e5f1-44be-8f00-5c3c628fff32",
	"name": "User1",
	"layer": "",
	"ens": "",
	"active": true,
	"transform": {
		"RotationID": 26,
		"ScaleID": 1,
		"PositionID": 60,
		"position": {
			"ID": 60,
			"CreatedAt": "2023-10-23T01:53:26.866887889Z",
			"UpdatedAt": "2023-10-23T01:53:26.866887889Z",
			"DeletedAt": null,
			"x": 1,
			"y": 2,
			"z": 3
		},
		"rotation": {
			"ID": 26,
			"CreatedAt": "2023-10-23T01:53:26.901747833Z",
			"UpdatedAt": "2023-10-23T01:53:26.901747833Z",
			"DeletedAt": null,
			"x": 4,
			"y": 5,
			"z": 6,
			"w": 0
		},
		"scale": {
			"ID": 1,
			"CreatedAt": "2023-10-23T01:53:26.973717247Z",
			"UpdatedAt": "2023-10-23T01:53:26.973717247Z",
			"DeletedAt": null,
			"x": 7,
			"y": 8,
			"z": 9
		}
	},
	"states": [
		{
			"ID": 1,
			"CreatedAt": "2023-10-23T01:53:27.011892644Z",
			"UpdatedAt": "2023-10-23T01:53:27.011892644Z",
			"DeletedAt": null,
			"PlayerID": 22,
			"stateID": 0,
			"value": 0.4
		},
		{
			"ID": 2,
			"CreatedAt": "2023-10-23T01:53:27.011892644Z",
			"UpdatedAt": "2023-10-23T01:53:27.011892644Z",
			"DeletedAt": null,
			"PlayerID": 22,
			"stateID": 0,
			"value": 0.1
		}
	]
}
```

**Review by Balugrizzly — APPROVED** [review]
Looks good

**Comment by Balugrizzly** [comment]
Is this ready for merge in your opinion?

**Comment by Sarahtonein** [comment]
> Is this ready for merge in your opinion?

Yes I think this is good to merge now :)

**Comment by Sarahtonein** [comment]
@Balugrizzly
